### ISSUE TITLE
Challenge Listing: No challenges found message displayed before challenges are loaded

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4376]
Challenge Listing: No challenges found message displayed before challenges are loaded
[challenge loading.mov.zip](https://github.com/topcoder-platform/community-app/files/4650473/challenge.loading.mov.zip)

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126385 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126385 has been updated - it has been assigned to thomaskranitsas.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126385 has been updated - sr_jr has been unassigned.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126385 has been updated - sr_jr has been unassigned.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126385 has been updated - sr_jr has been unassigned.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126385 has been updated - it has been assigned to sr_jr.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by simranb86** [comment]
hi @sushilshinde the code looks fine here, I was able to replicate the issue like 6-8 hrs back when working on this, but not now, looks like it was backend data issue.  

Please see this - 
[Issue_4376.mov.zip](https://github.com/topcoder-platform/community-app/files/4672991/Issue_4376.mov.zip)

**Comment by sushilshinde** [comment]
Sure, thanks @simranb86

**Comment by sushilshinde** [comment]
@SathyaJayabal can you please confirm

**Comment by simranb86** [comment]
> @SathyaJayabal can you please confirm

@SathyaJayabal  had a chance to re-verify this ?

**Comment by SathyaJayabal** [comment]
@simranb86 , Yes, I too do not see the message now before loading. Now the first challenge is displayed first and the rest of the challenges display after a couple of seconds. (like in your video)

**Comment by lakshmiathreya** [comment]
The lag is not acceptable:
[ChallengeLoading25May.mov.zip](https://github.com/topcoder-platform/community-app/files/4675240/ChallengeLoading25May.mov.zip)

**Comment by lakshmiathreya** [comment]
should we close this and log a new issue for the Lag?

**Comment by SathyaJayabal** [comment]
@simranb86 , sometimes it just stops after loading the first challenges (happens rarely). 

<img width="1440" alt="Screenshot 2020-05-25 at 11 02 18 AM" src="https://user-images.githubusercontent.com/58783823/82781404-54400500-9e77-11ea-87f0-0e43a054da00.png">

**Comment by simranb86** [comment]
> @simranb86 , sometimes it just stops after loading the first challenge (happens rarely).
> 
> <img alt="Screenshot 2020-05-25 at 11 02 18 AM" width="1440" src="https://user-images.githubusercontent.com/58783823/82781404-54400500-9e77-11ea-87f0-0e43a054da00.png">


@SathyaJayabal what you see is right - it is part of the impl, it is not lag or bug, it is backend data issue if we need to fix it, as it is being sent to front end. We make several backend calls and first one is coming back with only this challenges which has  `registrationPhase` as open. If you debug and see the console you would see something like below in first api response:

```
"phases": [
        {
          "duration": 173400,
          "actualStartDate": "2018-05-02T05:38:25.728-0400",
          "scheduledStartDate": "2018-05-02T05:38:25.728-0400",
          "name": "Registration",
          "scheduledEndDate": "2018-05-04T09:48:25.728Z",
          **"isOpen": true**
        },
        {
          "duration": 2760,
          "actualStartDate": "2018-05-02T05:43:43.721-0400",
          "scheduledStartDate": "2018-05-02T05:43:43.721-0400",
          "name": "Submission",
          "actualEndDate": "2018-05-02T06:29:26.977-0400",
          "scheduledEndDate": "2018-05-02T10:29:43.721Z",
          "isOpen": false
        },
        {
          "duration": 601740,
          "scheduledStartDate": "2018-05-02T06:29:26.977-0400",
          "name": "Review",
          "scheduledEndDate": "2018-05-09T09:38:26.977Z",
          "isOpen": false
        },
        {
          "duration": 86400,
          "scheduledStartDate": "2018-05-09T05:38:00.000-0400",
          "name": "Appeals",
          "scheduledEndDate": "2018-05-10T09:38:00.000Z",
          "isOpen": false
        },
        {
          "duration": 43200,
          "scheduledStartDate": "2018-05-10T05:38:00.000-0400",
          "name": "Appeals Response",
          "scheduledEndDate": "2018-05-10T21:38:00.000Z",
          "isOpen": false
        }
      ],
```

Or try this API call directly - https://api.topcoder-dev.com/v5/challenges/?status=Active&perPage=50&page=1

**Comment by sushilshinde** [comment]
@simranb86 why it not showing table placement holder animation?

Can we fix that?

**Comment by simranb86** [comment]
> @simranb86 why it doesnt continue to show table placement holder animation?
> 
> Can we fix that?

Yes, i think we can. Let me address this in end - once other tickets are done.

**Comment by Oanh-and-only-Oanh** [comment]
hi @simranb86, FYI this ticket has been reassigned to P1.

**Comment by simranb86** [comment]
> hi @simranb86, FYI this ticket has been reassigned to P1.

ok taking this on priority, thanks for flagging.

**Comment by SathyaJayabal** [comment]
@simranb86 , can  we have an ETA for this ticket please?

**Comment by simranb86** [comment]
> @simranb86 , can we have an ETA for this ticket please?

discussed with @sushilshinde  , waiting for direction we take to solve this.

**Comment by sushilshinde** [comment]
if we reduce calls then this issue will be resolved automatically. 

https://github.com/topcoder-platform/community-app/issues/4384#issuecomment-636970584

Just three calls, we can chain them in promises

**Comment by sushilshinde** [comment]
Just wait for My Challenges bucket to finish. Till that time waiting placeholder animation should display.

**Comment by simranb86** [comment]
> Placeholder animation should continue to show until My Challenges are loaded.
> 
> For a new member, My Challenges will be empty, so it will be Open for registration bucket.

Thanks, working on it

**Comment by simranb86** [comment]
@sushilshinde 
PR
- https://github.com/topcoder-platform/community-app/pull/4486
- https://github.com/topcoder-platform/topcoder-react-lib/pull/182

**Comment by simranb86** [comment]
Also as discussed over slack, bumped up perPage from 50 to 99 now.

**Comment by sushilshinde** [comment]
ok. thanks will review and merge

**Comment by SathyaJayabal** [comment]
verified on beta and test.
[4376.mov.zip](https://github.com/topcoder-platform/community-app/files/4744011/4376.mov.zip)

**Comment by simranb86** [comment]
@Oanh-and-only-Oanh @SathyaJayabal should we close this as it has been in QA Pass since 2 days now ?

**Comment by Oanh-and-only-Oanh** [comment]
@lakshmiathreya @SathyaJayabal, please close ticket if it's passed on Beta.

**Comment by SathyaJayabal** [comment]
Waiting for tcx_accepted label @sushilshinde, we can close this.
